src/provider/systemdmanager/service.rs

```diff
@@ -49,34 +63,60 @@ impl SystemdService {
         self.file.clone()
     }
 
-    /// Checks if the ActiveState is set to active.
-    pub async fn is_running(&self) -> anyhow::Result<bool> {
-        self.unit_proxy
-            .active_state()
-            .await
-            .map(|state| state == ActiveState::Active)
-            .map_err(|error| {
-                anyhow!(
-                    "ActiveState of systemd unit [{}] cannot be retrieved: {}",
-                    self.file,
-                    error
-                )
-            })
+    /// Returns a coarse-grained state of the service unit object.
+    ///
+    /// It is assumed that RemainAfterExit is set to "yes" in the given
+    /// unit if the service can terminate. Otherwise it would not be
+    /// possible to distinguish between "inactive and never run" and
+    /// "inactive and terminated successfully".
```

Comment:
Do we need to consider how a server restart would affect this?
I assume that after a server restart we cannot see the unit anymore?

Reply:
I tested it and a unit which was in state `active (exited)`, was in state `inactive (dead)` after a reboot. So if the machine is rebooted after a job terminated but before the Stackable Agent had the chance to delete it then we have an undesirable state. I will think about it if this problem can be solved.